kong/router.lua

```diff
@@ -1304,10 +1304,13 @@ function _M.new(routes)
   -- all routes indexed by id
   local routes_by_id = {}
 
+  if not cache then
```

Comment:
`cache` can be now given as function argument. If it is not given, it will fallback to old default, which was inside this branch. Many of our tests don't pass cache.